src/datachain/cli.py

```diff
@@ -789,25 +841,55 @@ def ls_remote(
 def ls(
     sources,
     long: bool = False,
-    remote: str = "",
-    config: Optional[Mapping[str, str]] = None,
+    studio: bool = False,
+    local: bool = False,
+    all: bool = False,
+    team: Optional[str] = None,
     **kwargs,
 ):
-    if config is None:
-        from .config import Config
+    token = Config().read().get("studio", {}).get("token")
```

Comment:
[I] Both `ls` and `datasets` now run the same logic to determine whether or not to call the local or studio flavour of their underlying commands. Can you pull-out the duplication, please?

[C] I would also say that IMO the default should be to reach out to Studio only if the `studio` or `all` flag is passed. Forget about the auto-token magic. 

I would also say that if `all` is passed that we want to display a table instead of two lists:

| name | version | studio | local |
|--|--|--|--|
| cats | 1 | ✅ | ❌ |
| cats | 2 | ✅ | ❌ | 
| dogs | 1 | ✅ | ✅ | 

That way I can easily check the discrepancies between the two. Instead of manually having to compare the two lists.